cmd/translate/main_grpc_integration_test.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	pb "go.expect.digital/translate/pkg/server/translate/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+)
+
+func TestMain(m *testing.M) {
+	go main()
+
+	// Needed to prevent panic, when service is not ready to receive requests
+	time.Sleep(time.Millisecond * 50)
+
+	os.Exit(m.Run())
+}
+
+func createConnection(ctx context.Context, t *testing.T) *grpc.ClientConn {
+	t.Helper()
+
+	conn, err := grpc.DialContext(
+		ctx,
+		"localhost:8080",
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	if err != nil {
+		log.Panic("Failed to dial bufnet: %w", err)
+	}
+
+	return conn
+}
+
+func Test_UploadTranslationFile_gRPC(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	type args struct {
+		req *pb.UploadTranslationFileRequest
+	}
+
+	tests := []struct {
+		args args
+		name string
+		want codes.Code
+	}{
+		{
+			name: "Happy path",
+			args: args{
+				req: &pb.UploadTranslationFileRequest{
+					Language: "lv-lv",
+					Data: []byte(`{
+						"language":"lv-lv",
+						"messages":[
+							 {
+									"id":"1",
+									"meaning":"When you great someone",
+									"message":"hello",
+									"translation":"čau",
+									"fuzzy":false
+							 }
+						]
+				 }`),
+				},
+			},
+			want: codes.OK,
+		},
+		{
+			name: "Missing language",
+			args: args{
+				req: &pb.UploadTranslationFileRequest{
+					Data: []byte(`{
+						"messages":[
+							 {
+									"id":"1",
+									"meaning":"When you great someone",
+									"message":"hello",
+									"translation":"čau",
+									"fuzzy":false
+							 }
+						]
+				 }`),
+				},
+			},
+			want: codes.InvalidArgument,
+		},
+		{
+			name: "Missing data",
+			args: args{
+				req: &pb.UploadTranslationFileRequest{
+					Language: "lv-lv",
+				},
+			},
+			want: codes.InvalidArgument,
+		},
+		{
+			name: "Invalid language",
+			args: args{
+				req: &pb.UploadTranslationFileRequest{
+					Language: "xyz-ZY-Latn",
+					Data: []byte(`{
+						"messages":[
+							 {
+									"id":"1",
+									"meaning":"When you great someone",
+									"message":"hello",
+									"translation":"čau",
+									"fuzzy":false
+							 }
+						]
+				 }`),
+				},
+			},
+			want: codes.InvalidArgument,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			conn := createConnection(ctx, t)
```

Comment:
Do you have to create a connection for each test?

Reply:
My aim intention to increase its independence.
Each request is initiated by a new connection and client, but I'm not sure if that's more accurate or superior to having a single client and connection to test all endpoints.